### PR TITLE
LibJS: Math built-in improvements

### PR DESCRIPTION
**Pull request by IdanHo**
This PR includes a myriad of fixes and additions to Math.prototype methods that allows (together with #7796) all but one (`built-ins/Math/round/S15.8.2.15_A6`) Math built-in related test262 cases to now pass.

The last commit is a more general fix that fixes every `Symbol.toStringTag` test262 test including the Math built-in one.

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/ObjectPrototype.cpp [hunk 1]
```diff
@@ -111,13 +111,13 @@ JS_DEFINE_NATIVE_FUNCTION(ObjectPrototype::value_of)
 
 JS_DEFINE_NATIVE_FUNCTION(ObjectPrototype::property_is_enumerable)
 {
-    auto name = vm.argument(0).to_string(global_object);
+    auto property_key = StringOrSymbol::from_value(global_object, vm.argument(0));
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/ObjectPrototype.cpp:114]: Oh look, you found what my recent meme was about :D

Sadly this is still not fully spec compliant, as `from_value()` differs from `ToPropertyKey()` in subtle ways - still better than ignoring symbols altogether though, I'll take it :)

**Review by linusg — COMMENTED** [review]
Typo in 5th commit: `encounetered`

File: Userland/Libraries/LibJS/Runtime/ObjectPrototype.cpp (hunk 1 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/ObjectPrototype.cpp:114]: too bad, ill look into it a bit more tommorow then

**Comment by IdanHo** [comment]
typo fixed :)

File: Userland/Libraries/LibJS/Runtime/MathObject.cpp [hunk 2]
```diff
@@ -428,6 +429,17 @@ JS_DEFINE_NATIVE_FUNCTION(MathObject::hypot)
...
 
+JS_DEFINE_NATIVE_FUNCTION(MathObject::imul)
+{
+    auto a = vm.argument(0).to_u32(global_object);
+    if (vm.exception())
+        return {};
+    auto b = vm.argument(1).to_u32(global_object);
+    if (vm.exception())
+        return {};
+    return Value(static_cast<i32>(a * b));
```
Comment (awesomekling) [inline Userland/Libraries/LibJS/Runtime/MathObject.cpp]: This looks like it would fail if either of the inputs were a negative number.

Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/MathObject.cpp]: I'm just following the specification: https://tc39.es/ecma262/#sec-math.imul

Comment (awesomekling) [reply Userland/Libraries/LibJS/Runtime/MathObject.cpp]: Oh, okay! :)

**Comment by IdanHo** [comment]
Setting this to draft until #7811 is merged so i can update this to use the new API :)

**Comment by IdanHo** [comment]
This should be ready to merge as well :)

**Comment by linusg** [comment]
was just waiting for CI to finish :) Not the most straightforward code, but nice work!

**Pull request merged**